lib/drivers/abbott/abbottFreeStyleNeo.js

```diff
@@ -0,0 +1,415 @@
+          .with_value(value)
+          .with_units('mmol/L') // values are always in 'mmol/L'
+          .with_deviceTime(sundial.formatDeviceTime(jsDate))
+          .set('index', index);
+
+        cfg.tzoUtil.fillInUTCInfo(recordBuilder, jsDate);
+
+        if (annotation) {
+          annotate.annotateEvent(recordBuilder, annotation);
+        }
+
+        const postRecord = recordBuilder.done();
+        delete postRecord.index;
+        data.post_records.push(postRecord);
+      } else {
+        debug('Skipping ketone control solution test');
+      }
+    });
+  }
+
+  return {
+    /* eslint no-param-reassign:
+       [ "error", { "props": true, "ignorePropertyModificationsFor": ["data"] } ] */
+
+    detect(deviceInfo, cb) {
+      debug('no detect function needed', deviceInfo);
+      cb(null, deviceInfo);
+    },
+
+    setup(deviceInfo, progress, cb) {
+      debug('in setup!');
+      progress(100);
+      cb(null, { deviceInfo });
+    },
+
+    connect(progress, data, cb) {
+      hidDevice.connect(cfg.deviceInfo, FreeStyleProtocol.probe, (err) => {
+        if (err) {
+          return cb(err);
+        }
+        return protocol.initCommunication(() => {
+          data.disconnect = false;
+          progress(100);
+          return cb(null, data);
+        });
+      });
+    },
+
+    getConfigInfo(progress, data, cb) {
+      progress(0);
+
+      (async () => {
+        cfg.deviceInfo.serialNumber = await driver.getSerialNumber();
+        cfg.deviceInfo.deviceId = `${cfg.deviceInfo.driverId}-${cfg.deviceInfo.serialNumber}`;
+        cfg.deviceInfo.deviceTime = sundial.formatDeviceTime(await driver.getDateTime());
+      })().then(() => {
+        common.checkDeviceTime(
+          cfg,
+          (timeErr, serverTime) => {
+            progress(100);
+            if (timeErr) {
+              if (timeErr === 'updateTime') {
+                cfg.deviceInfo.annotations = 'wrong-device-time';
+                const newTime = sundial.formatInTimezone(serverTime, cfg.timezone, 'HH,mm');
+                const newDate = sundial.formatInTimezone(serverTime, cfg.timezone, 'MM,DD,YY');
+
+                (async () => {
+                  await driver.setTime(newTime, newDate);
+                })().then(() => {
+                  data.connect = true;
+                  return cb(null, data);
+                }).catch((error) => {
+                  debug('Error in getConfigInfo: ', error);
+                  return cb(error, null);
+                });
+              } else {
+                cb(timeErr, null);
+              }
+            } else {
+              data.connect = true;
+              cb(null, data);
+            }
+          },
+        );
+      }).catch((error) => {
+        debug('Error in getConfigInfo: ', error);
+        return cb(error, null);
+      });
+    },
+
+    fetchData(progress, data, cb) {
+      debug('in fetchData', data);
+      (async () => {
```

Comment:
If you wanted to, you can remove the "then/catch" Promise handling with:
```
(async () => {
  try {
    data.records = await driver.getRecords();
    progress(100);
    return cb(null, data);
  } catch (error) {
    return cb(error, null);
  }
})()
```

Reply:
Yeah, I like that idea.